packages/CodeGenLib/src/sql_codegen.ts

```diff
@@ -833,8 +832,7 @@ GO${permissions}
               !ef.IsVirtual &&
               (!ef.IsPrimaryKey || !autoGeneratedPrimaryKey || isUpdate) &&
               ef.Name.toLowerCase().trim() !== 'updatedat' &&
-              ef.Name.toLowerCase().trim() !== 'createdat' &&
-              (ef.Type.toLowerCase().trim() !== 'uniqueidentifier' || ef.IsPrimaryKey)
```

Comment:
@AN-BC This change was to overcome issues arising from not being able to update FK columns that are typed as `uniqueidentifier` (although they should probably be typed varchar). There may be Unintended Consequences here– would love your opinion on the change

Reply:
@cadam11 the removal of `createdat` is one I would reconsider, that field name has special meaning and shouldn't be passed into spCreate. As for the inclusion of uniqueidentifier columns into the spCreate, I wonder if we can use the `AllowUpdateAPI` flag on the EntityField record to determine if we include this in spCreate. For situations where the developer/sysadmin wants to allow uniqueidentifiers to be created by the client code and passed in, we can allow them in, but if `AllowUpdateAPI===0` then we don't include it, instead of checking the Primarykey flag?

Reply:
This change doesn't remove `createdat`, just the `&& (ef.Type.toLowerCase().trim() !== 'uniqueidentifier' || ef.IsPrimaryKey)` clause.

It also does preserve the existing behavior for `AllowUpdateAPI` such that an entity field will only be included in the declaration if `AllowUpdateAPI` is set, or both `IsPrimaryKey` and `isUpdate` are true.

Sounds like this change is unlikely to introduce new, undesired behavior so I think we're ok to merge it.